pkg/server/sandbox_run_test.go

```diff
@@ -269,5 +286,36 @@ func TestRunPodSandbox(t *testing.T) {
 	assert.Equal(t, expectedPluginArgument, pluginArgument, "SetUpPod should be called with correct arguments")
 }
 
+func TestParseDNSOption(t *testing.T) {
+	for desc, test := range map[string]struct {
+		servers  []string
+		searches []string
+		options  []string
+	}{
+		"empty dns options": {},
+		"non-empty dns options": {
+			servers:  []string{"8.8.8.8", "server.google.com"},
+			searches: []string{"114.114.114.114"},
+			options:  []string{"timeout:1"},
+		},
```

Comment:
1) Let's add test case to check whether the function returns error when searches is more than 6
2) Let's add `expectedOutput` into the test case, and directly check `assert.Equal(t, expectedOutput, resolvContent)`, e.g.:
```golang
map[string]struct{
dnsConfig *runtime.DNSConfig
expectedContent string
expectErr bool
} {
 "empty dns options should return empty content": {
    dnsConfig: &runtime.DNSConfig{},
    expectedOutput: "",
  },
  "non-empty dns options should return correct content": {
   dnsConfig: &runtime.DNSConfig{/*...*/},
   expectedOutput: `search 114.114.114.114
nameserver 8.8.8.8
nameserver server.google.com
...
`
  },
  "should return error if dns search exceeds limit": {
   /*...*/
  },
}
```